File: 11.BuildBST/buildbst.java
import java.util.*;

class Node {
    int data;
    Node left, right;
    Node(int d) {
        data = d;
        left = right = null;
    }
}

public class ValidateBST {

    static Scanner sc = new Scanner(System.in);

    static Node buildTree() {
        if (!sc.hasNextInt()) return null;
        int val = sc.nextInt();
        if (val == -1) return null;

        Node node = new Node(val);
        node.left = buildTree();
        node.right = buildTree();
        return node;
    }

    static boolean isValidBST(Node root, long min, long max) {
        if (root == null) return true;
        if (root.data <= min || root.data >= max) return false;
        return isValidBST(root.left, min, root.data) &&
               isValidBST(root.right, root.data, max);
    }

    public static void main(String[] args) {
        Node root = buildTree();
        System.out.print(isValidBST(root, Long.MIN_VALUE, Long.MAX_VALUE) ? "True" : "False");
    }
}

File: 11.BuildBST/buildbst.py
values=list(map(int,input().split()))

class Node:
    def __init__(self,value):
        self.data=value 
        self.left=None 
        self.right=None 

root=None 

def insert(root,value):
    if root is None:
        root=Node(value)
    elif value<root.data:
        root.left=insert(root.left,value)
    else:
        root.right=insert(root.right,value)
    return root 

for i in values[:len(values)-1]:
    root=insert(root,i)

def inorder(root):
    if root:
        inorder(root.left)
        print(root.data,end=" ")
        inorder(root.right)
inorder(root)
